internal/api/fetcher/fetcher.go

```diff
@@ -46,6 +48,28 @@ type apiObject interface {
 	URL() (string, error)
 }
 
+// FetchFromURL fetches the content from the given URL and returns the content as a byte slice.
+func FetchFromURL(ctx context.Context, client HTTPClient, sourceURL *url.URL) ([]byte, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, sourceURL.String(), http.NoBody)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return []byte{}, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return []byte{}, fmt.Errorf("http status code: %d", resp.StatusCode)
+	}
+	content, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return []byte{}, err
+	}
+	return content, nil
```

Comment:
```suggestion
	return io.ReadAll(resp.Body)
```
Same as above: If there is no additional error wrapping, just return the output of the function directly.